### PR TITLE
Add council tax payer view endpoint

### PR DESCRIPTION
**Pull request by Renny4Real**
To Do:

- [x] Add controller endpoint and test(s)
- [x] Add End to End test(s) for taxpayer view endpoint
- [x] Check if a tax payer's email should be returned as a list or as a string (Check DB? Initial assumption was that there were multiple emails that could be saved but looking at the `syemail` table implies otherwise)
- [ ] Refactor & clean up code.
- [ ] Set up a canary + any other appropriate alerting

*Open to feedback, comments, suggestions* ☺️

### CODE REVIEW

File: AcademyResidentInformationApi.Tests/V1/Helper/TestHelper.cs [hunk 1]
```diff
@@ -56,5 +56,63 @@ public static Claim CreateDatabaseClaimEntity(int? claimId)
...
+        public static CouncilProperty CreateDatabasePropertyForTaxPayer(string propertyRef)
+        {
+            var faker = new Fixture();
+            var cp = faker.Build<CouncilProperty>()
+                .With(p => p.PropertyRef, propertyRef)
+                .Create();
+            return cp;
+        }
+
+        public static Occupation CreateDatabaseOccupationEntityForCouncilProperty(int accountRef)
```
Comment (LBHBReynolds-Carr) [inline AcademyResidentInformationApi.Tests/V1/Helper/TestHelper.cs:80]: Have we created the occupation item to contain account ref and person number? Why did we call kit occupation?

File: AcademyResidentInformationApi.Tests/V1/Controllers/TaxPayersControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using AutoFixture;
+using NUnit.Framework;
+
+namespace AcademyResidentInformationApi.Tests.V1.Controllers
+{
+    [TestFixture]
+    public class TaxPayersControllerTests
+    {
+        private TaxPayersController _classUnderTest;
+        private Mock<IGetTaxPayerByIdUseCase> _mockGetTaxPayerByIdUseCase;
```
Comment (LBHBReynolds-Carr) [inline AcademyResidentInformationApi.Tests/V1/Controllers/TaxPayersControllerTests.cs:17]: We're only testing for Getting a Tax Payer for ID, should we not also be testing for getting all tax payers?

**Comment by humulla** [comment]
Would you not need a context for academy tax or have I just missed that part?

**Comment by elcorbs** [comment]
> Would you not need a context for academy tax or have I just missed that part?

The tables are in the same database so AcademyContext can be used for both.

File: AcademyResidentInformationApi/V1/Infrastructure/Occupation.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace AcademyResidentInformationApi.V1.Infrastructure
+{
+    [Table("ctoccupation", Schema = "dbo")]
```
Comment (LBHBReynolds-Carr) [inline AcademyResidentInformationApi/V1/Infrastructure/Occupation.cs:6]: Why are we using the word "occupation" for this element?

**Review by LBHBReynolds-Carr — COMMENTED** [review]
Looks good

File: AcademyResidentInformationApi.Tests/V1/Controllers/TaxPayersControllerTests.cs (hunk 2 above)
Comment (elcorbs) [reply AcademyResidentInformationApi.Tests/V1/Controllers/TaxPayersControllerTests.cs:17]: This piece of work was for get by Id functionality. when the list endpoint is developed that test will be written.

File: AcademyResidentInformationApi/V1/Infrastructure/Occupation.cs (hunk 3 above)
Comment (elcorbs) [reply AcademyResidentInformationApi/V1/Infrastructure/Occupation.cs:6]: This is the name of the table in Academy

File: AcademyResidentInformationApi.Tests/V1/Helper/TestHelper.cs (hunk 1 above)
Comment (elcorbs) [reply AcademyResidentInformationApi.Tests/V1/Helper/TestHelper.cs:80]: This name comes from the academy database but I think it relates to the occupation of the house, so the people who live there

**Comment by LBHBomotoso** [comment]
Looks fine

**Review by LBHBReynolds-Carr — APPROVED** [review]

**Pull request merged**